### PR TITLE
EVG-6080: add command to download and extract files on hosts

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/EVG-6080

* `evergreen host download` command can download and optionally extract files.

### CODE REVIEW

File: operations/host.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ func Host() cli.Command {
 		Usage: "manage evergreen spawn and build hosts",
 		Subcommands: []cli.Command{
 			hostCreate(),
+			hostDownload(),
```
Comment (tychoish) [inline operations/host.go]: does it make more sense to do this in the agent command hierarchy?

File: operations/host_download.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+			return nil
+		},
+	}
+}
+
+// downloadAndExtract downloads the archive file from url as a temporary
+// file, extracts the archive file to the directory destDir, and removes the
+// temporary archive.
+func downloadAndExtract(url, destDir string) error {
```
Comment (tychoish) [inline operations/host_download.go]: doesn't jasper have support for this kind of operation? does it make more sense to delegate to that code?

File: config.go [hunk 3]
```diff
@@ -25,7 +25,7 @@ var (
 	BuildRevision = ""
 
 	// Commandline Version String; used to control auto-updating.
-	ClientVersion = "2019-04-04"
+	ClientVersion = "2019-04-11"
```
Comment (tychoish) [inline config.go]: given that this is internal, I don't think we need to force people to upgrade.

File: operations/host_download.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+			if !extract && len(fileNames) != 1 {
+				return errors.New("must specify exactly one file name if not extracting")
+			}
+
+			filePaths := make([]string, 0, len(fileNames))
+			for _, fileName := range fileNames {
+				filePaths = append(filePaths, filepath.Join(destDir, fileName))
+			}
+
+			if err := os.MkdirAll(destDir, 0755); err != nil {
```
Comment (bsamek) [inline operations/host_download.go]: Should this validate that `destDir` is set?

**Comment by Kimchelly** [comment]
I'm closing this PR and the ticket - I was hoping to re-use this functionality between SSH and user data bootstrapping. However, this doesn't seem possible since the evergreen binary won't be on the machine by the time it's bootstrapped with user data.

**Pull request closed without merging**